src/clj_irods/validate.clj

```diff
@@ -0,0 +1,67 @@
+(ns clj-irods.validate
+  (:require [clojure-commons.error-codes :as error]
+            [clojure.tools.logging :as log]
+            [slingshot.slingshot :refer [throw+]]
+            [clj-jargon.permissions :as jargon-perms]
+            [clj-irods.core :as rods]))
+
+(defn validate
+  "Validate a set of things in iRODS.
+
+  Each validation is a vector of a keyword (identifying the kind of validation)
+  and any relevant arguments. The validations will be run in the provided order.
+
+  Available validations and their arguments:
+
+  :user-exists (string or vector, users to check), (string zone)
+  :path-exists (string or vector, path or paths to check), (string user), (string zone)
+  :path-is-file (string or vector, path or paths to check), (string user), (string zone)
+  :path-is-dir (string or vector, path or paths to check), (string user), (string zone)
+  :path-readable (string or vector, path or paths to check), (string user), (string zone)
+  :path-writeable (string or vector, path or paths to check), (string user), (string zone)
+  :path-owned (string or vector, path or paths to check), (string user), (string zone)
+  "
+  [irods & validations]
+
+  (doseq [v validations]
+    (condp = (first v)
```

Comment:
If anyone has some good ideas on how to make the below stuff a bit less repetitive/verbose, that might be good. I could probably remove the repetition of `rest` by destructuring in the `doseq`, but I'd still need to destructure the specific arguments for a given validation, so it wouldn't change all that much. I couldn't think of anything particularly clever for the vector-vs-single-item code either.

It may just need to be verbose, but anyway.